### DISCUSSION TITLE
Getting Error on The Certificate

### DISCUSSION THREAD

**Discussion by gnp2308** [jpos/jPOS-EE#246] [General]
Hi Team,

I am new to jposee. I just downloaded the jposee. I am trying to execute gradle build for jposee. It is giving an error. Can someone help. Details posted below :

---------------

jPOS-EE$ gradle install
Parallel execution is an incubating feature.

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'jposee'.
> Could not resolve all files for configuration ':classpath'.
   > Could not resolve org.owasp:dependency-check-gradle:6.1.5.
     Required by:
         project :
      > Could not resolve org.owasp:dependency-check-gradle:6.1.5.
         > Could not get resource 'https://repo.maven.apache.org/maven2/org/owasp/dependency-check-gradle/6.1.5/dependency-check-gradle-6.1.5.pom'.
            > Could not HEAD 'https://repo.maven.apache.org/maven2/org/owasp/dependency-check-gradle/6.1.5/dependency-check-gradle-6.1.5.pom'.
               > PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s

-----------------------

Regards,
Ganesh

**Comment by ar** [comment]
I suggest you use `./gradlew` (the Gradle wrapper provided by jPOS-EE) so that you can make sure you are using the same Gradle version we use to build jPOS-EE. That said, if you are not going to make changes to jPOS-EE itself, you don't have to build it locally. Just use the builds provided at https://jpos.org/maven

**Reply by gnp2308** [reply]
Hi Ar,

I tried using the above option, but here the gradle daemon fails to start. Snapshot below :
-------------------------------------------------
/mnt/f/jPOS-EE$ ./gradlew install
Starting a Gradle Daemon, 13 busy Daemons could not be reused, use --status for details
Starting a Gradle Daemon, 14 busy Daemons could not be reused, use --status for details
Starting a Gradle Daemon, 15 busy Daemons could not be reused, use --status for details
Starting a Gradle Daemon, 16 busy Daemons could not be reused, use --status for details

> Starting Daemon
-----------------------------------------------------

I tried the same command with sudo user. Here the daemon starts , but the same error of ssl appears. snapshot below:

-------------------------
sudo ./gradlew install
[sudo] password for xxxx:
Downloading https://services.gradle.org/distributions/gradle-6.8.2-bin.zip
..........10%..........20%..........30%...........40%..........50%..........60%..........70%...........80%..........90%..........100%

Welcome to Gradle 6.8.2!

Here are the highlights of this release:
 - Faster Kotlin DSL script compilation
 - Vendor selection for Java toolchains
 - Convenient execution of tasks in composite builds
 - Consistent dependency resolution

For more details see https://docs.gradle.org/6.8.2/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'jposee'.
> Could not resolve all artifacts for configuration ':classpath'.
   > Could not resolve org.owasp:dependency-check-gradle:6.1.5.
     Required by:
         project :
      > Could not resolve org.owasp:dependency-check-gradle:6.1.5.
         > Could not get resource 'https://repo.maven.apache.org/maven2/org/owasp/dependency-check-gradle/6.1.5/dependency-check-gradle-6.1.5.pom'.
            > Could not GET 'https://repo.maven.apache.org/maven2/org/owasp/dependency-check-gradle/6.1.5/dependency-check-gradle-6.1.5.pom'.
               > The server may not support the client's requested TLS protocol versions: (TLSv1.2, TLSv1.3). You may need to configure the client to allow other protocols to be used. See: https://docs.gradle.org/6.8.2/userguide/build_environment.html#gradle_system_properties
                  > PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 19s
-----------------------

Regards,
Ganesh

**Reply by ar** [reply]
Perhaps you're running behind a firewall, without access to Maven Central.

**Reply by gnp2308** [reply]
Hi AR,

Thanks for the support. I was able to resolve the PKIX path error by doing the following :
1.  Added in gradle.properties: 
org.gradle.jvmargs=-Dfile.encoding=UTF-8 -Djavax.net.ssl.keyStore={path to cacerts} -Djavax.net.ssl.keyStorePassword={password for keystore}
2. Imported the client crt/pem file for maven in the cacerts 
keytool -import -file ./repo1-maven-org.pem -keystore {path to cacerts}

Now the command is working atleast with sudo (sudo ./gradlew) . Let me see the results. Thanks for assistance.

Regards,
Ganesh
